### ISSUE TITLE
Gradients Explode when Training with Qwen and its Node Embeddings

### ISSUE THREAD

**Issue by ammar-n-abbas** [dinobby/MAGDi#4]
Hi,

I tried to fine-tune Qwen2.5-7B-Instruct with MAGDi training pipeline; however, the gradients explode to Nan right after the first training epoch. Can it be related to node embeddings not being normalized?

Thank you
